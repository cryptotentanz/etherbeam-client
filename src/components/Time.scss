@use '../styles/resources/fonts';

.my-time {
  font-family: fonts.$monospace;
}
