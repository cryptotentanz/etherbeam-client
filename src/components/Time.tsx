import moment from 'moment'
import React, { FC } from 'react'
import './Time.scss'

interface TimeProps {
  dateTime: Date
}

const Time: FC<TimeProps> = ({ dateTime }: TimeProps) => {
  return <span className="my-time">{moment(dateTime).format('HH:mm')}</span>
}

export default Time
