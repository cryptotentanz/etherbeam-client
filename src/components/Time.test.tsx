import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Time from './Time'

describe('<Time />', () => {
  const dateTime = new Date(2020, 4, 5, 11, 22, 33)
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Time dateTime={dateTime} />)
    })

    it('should render', () => {
      const span = component.find('span').at(0)
      expect(span.hasClass('my-time')).toBeTruthy()
      expect(span.prop('children')).toBe('11:22')
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })
})
