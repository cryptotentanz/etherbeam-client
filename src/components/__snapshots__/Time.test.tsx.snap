// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Time /> Default should match snapshot 1`] = `
<span
  class="my-time"
>
  11:22
</span>
`;
